Userland/Libraries/LibJS/Contrib/Test262/262Object.cpp

```diff
@@ -62,13 +62,16 @@ JS_DEFINE_NATIVE_FUNCTION($262Object::clear_kept_objects)
 
 JS_DEFINE_NATIVE_FUNCTION($262Object::create_realm)
 {
-    auto realm = MUST_OR_THROW_OOM(Realm::create(vm));
-    auto realm_global_object = vm.heap().allocate_without_realm<GlobalObject>(*realm);
-    VERIFY(realm_global_object);
-    realm->set_global_object(realm_global_object, nullptr);
-    set_default_global_bindings(*realm);
-    realm_global_object->initialize(*realm);
-    return Value(realm_global_object->$262());
+    JS::GCPtr<JS::Test262::GlobalObject> global_object;
+    auto root_execution_context = MUST(JS::Realm::initialize_host_defined_realm(
+        vm,
+        [&](JS::Realm& realm) -> JS::GlobalObject* {
+            global_object = vm.heap().allocate_without_realm<JS::Test262::GlobalObject>(realm);
+            return global_object;
+        },
+        nullptr));
+    vm.pop_execution_context();
```

Comment:
This sticks out to me as something we weren't doing before - are we supposed to do this pop?

When I run test262 with this PR I get bad results:
```
» ./per_file_result_diff.py --old ~/Desktop/per-file-master.json --new ~/Desktop/dev.json -i
    Diff Tests:
         -5677 ✅   +5677 ❌
```

Reply:
Hmm, that's weird, it's to do with the push done here: https://github.com/LadybirdBrowser/ladybird/pull/2116/files#diff-227de7b77ebe45bef391b3f60ab998587f0fa0b4ef24b84d2ceb654c1b9902c3R47 🤔 

The push wasn't happening as far as I can tell from the previous steps

I'll dig into one of the failing tests, that's a bit funky. Another option might be to expose some extra setters on `Realm` and do some of this stuff manually. It didn't all into a heap about it

Reply:
I'm not completely sure that this pop is wrong, I just saw the test failures and this was the one thing that immediately jumped out to me. The failure could totally be somewhere else entirely 🙃

Reply:
The pop ended up being fine, it was just a silly goof where I had accidentally missed the very last step of the realm initialization that performed host-defined initialization properties of the global object.